### DISCUSSION TITLE
dev: Enforce `unwrap` dev guideline with clippy

### DISCUSSION THREAD

**Discussion by epage** [GitoxideLabs/gitoxide#959] [Q&A] [answered]
> handle all errors, never unwrap(). If needed, expect("why").

This can be enforced with clippy:
```diff
diff --git i/.cargo/config.toml w/.cargo/config.toml
index 3e6810586..abd2a1e15 100644
--- i/.cargo/config.toml
+++ w/.cargo/config.toml
@@ -10,6 +10,7 @@ rustflags = [
     "-W", "clippy::redundant_closure_for_method_calls",
     "-W", "clippy::unnested_or_patterns",
     "-W", "clippy::uninlined_format_args",
+    "-W", "clippy::unwrap_used",

     # Rejected for now, and why
     # "-W" "clippy::default_trait_access" - sometimes makes imports necessary, just for a default value. It's good for more explicit typing though.
diff --git i/clippy.toml w/clippy.toml
index e034672c7..b824a53e3 100644
--- i/clippy.toml
+++ w/clippy.toml
@@ -1 +1,2 @@
 msrv = "1.65.0"
+allow-unwrap-in-tests = true
```

I would create a PR but there are a lot of existing unwrap calls, in particular from what look like test libraries but also production code.  Thought I'd mention this in case there wasn't awareness and interest by someone in taking this on

**Comment by Byron** [comment, marked as answer]
Thanks a lot for the hint, I wasn't aware!

I have added the lint to [the clippy tracking issue](https://github.com/Byron/gitoxide/issues/874). Do you happen to recall how many errors there were?

**Reply by epage** [reply]
Looks to be 172
